metadata-ingestion/src/datahub/ingestion/source/looker/lookml_source.py

```diff
@@ -1040,5 +1046,63 @@ def report_skipped_unreachable_views(
                     context=(f"Project: {project}, View File Path: {path}"),
                 )
 
+    def _optimize_views_by_common_explore(
+        self,
+        view_to_explores: Dict[str, Set[str]],
+        explore_to_views: Dict[str, Set[str]],
+    ) -> Dict[str, str]:
+        """
+        Optimize view-to-explore mapping by grouping views to minimize API calls.
+
+        This uses a greedy algorithm that prioritizes explores that appear in the most views,
+        maximizing the number of views assigned to the same explore.
+
+        Args:
+            view_to_explores: Dict mapping view_name -> set of explore_names
+            explore_to_views: Dict mapping explore_name -> set of view_names
+
+        Returns:
+            Dict mapping view_name -> explore_name (optimized assignment)
+        """
+
+        # Pre-compute explore sizes
+        explore_sizes = {
+            explore: len(views) for explore, views in explore_to_views.items()
+        }
+
+        # Build view-to-explore mapping using dynamic programming approach
+        view_to_explore: Dict[str, str] = {}
+
+        # For each view, find the explore with maximum size that contains it
+        for view_name in view_to_explores:
+            # Get all explores that contain this view from pre-built mapping
+            candidate_explores = view_to_explores[view_name]
```

Comment:
isn't this the same?

```suggestion
        # For each view, find the explore with maximum size that contains it
        for view_name, candidate_explores in view_to_explores.items():
```

or we cannot do that because we are updating `view_to_explore[view_name]` in the same iteration?

Reply:
Updated!